dhis-2/dhis-services/dhis-service-validation/src/main/java/org/hisp/dhis/outlierdetection/util/OutlierDetectionUtils.java

```diff
@@ -27,50 +27,86 @@
+  public static final String ERR_MSG_SILENT_FALLBACK =
+      "An exception occurred - silently fallback since it's multiple analytics query: ";
+
   /**
    * Returns an organisation unit 'path' "like" clause for the given list of {@link
    * OrganisationUnit}.
    *
-   * @param query the list of {@link OrganisationUnit}.
+   * @param orgUnits the list of {@link OrganisationUnit}.
    * @return an organisation unit 'path' "like" clause.
    */
-  public static String getOrgUnitPathClause(List<OrganisationUnit> orgUnits) {
-    String sql = "(";
+  public static String getOrgUnitPathClause(List<OrganisationUnit> orgUnits, String pathAlias) {
+    StringBuilder sql = new StringBuilder("(");
+    orgUnits.forEach(
+        ou ->
+            sql.append(pathAlias).append(".\"path\" like '").append(ou.getPath()).append("%' or "));
 
-    for (OrganisationUnit ou : orgUnits) {
-      sql += "ou.\"path\" like '" + ou.getPath() + "%' or ";
-    }
-
-    return StringUtils.trim(TextUtils.removeLastOr(sql)) + ")";
+    return StringUtils.trim(TextUtils.removeLastOr(sql.toString())) + ")";
   }
 
   /**
-   * Returns a period data start date clause.
+   * Wraps the provided interface around a common exception handling strategy.
    *
-   * @param dataStartDate the data start date.
-   * @return a period data start date clause.
+   * @param supplier the {@link Supplier} containing the code block to execute and wrap around the
+   *     exception handling.
+   * @return the {@link Optional} wrapping th result of the supplier execution.
    */
-  public static String getDataStartDateClause(Date dataStartDate) {
-    return dataStartDate != null ? "and pe.startdate >= :data_start_date " : StringUtils.EMPTY;
+  public static <T> Optional<T> withExceptionHandling(Supplier<T> supplier) {
+    try {
+      return Optional.ofNullable(supplier.get());
+    } catch (BadSqlGrammarException ex) {
+      if (relationDoesNotExist(ex.getSQLException())) {
+        log.info(ERR_MSG_TABLE_NOT_EXISTING, ex);
+        throw ex;
+      }
+      log.info(ERR_MSG_SILENT_FALLBACK, ex);
+    } catch (QueryRuntimeException ex) {
+      log.error("Internal runtime exception", ex);
+      throw ex;
+    } catch (DataIntegrityViolationException ex) {
+      log.error(E2208.getMessage(), ex);
+      throw new IllegalQueryException(ErrorCode.E2208);
+    } catch (DataAccessResourceFailureException ex) {
+      log.error(E7131.getMessage(), ex);
+      throw new QueryRuntimeException(E7131);
+    }
+
+    return Optional.empty();
   }
 
-  /**
-   * Returns a period data end date clause.
-   *
-   * @param dataStartDate the data start date.
-   * @return a period data end date clause.
-   */
-  public static String getDataEndDateClause(Date dataStartDate) {
-    return dataStartDate != null ? "and pe.enddate <= :data_end_date " : StringUtils.EMPTY;
+  private static boolean relationDoesNotExist(SQLException ex) {
```

Comment:
Why duplicate the same method in `WebMessageUtils`?
It could not be the best place for that method. But it could be moved to a more appropriate class and reused by analytics code and also outliers.

Reply:
?it is possible to reuse it after small redesign...

Reply:
done